Assets/Scripts/R&D Tree/ResearchNode.cs

```diff
@@ -0,0 +1,134 @@
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.UI;
+
+public class ResearchNode : MonoBehaviour
+{
+    [SerializeField] private Button _button;
+    [SerializeField] private Slider _researchBar;
+
+    private float _timer;
+
+    public UnityEvent ResearchDoneEvent = new UnityEvent();
+
+    private ResearchNodeSetting _nodeSetting;
+
+    public SkillState CurrentSkillState;
+
+    private void OnEnable()
+    {
+        SetUIForStates();
+    }
+
+    private void Start()
+    {
+        _nodeSetting = GetComponent<ResearchNodeSetting>();
+
+        _researchBar.maxValue = _nodeSetting.ResearchTime;
+        AddListenersToEvent();
+    }
+
+    /// <summary>
+    /// This function is called when the researchnode started to be researched
+    /// </summary>
+    public void TimerForInDevelopment()
```

Comment:
When adding the value to the node, think about the rest value when a node needs less research value than is supplied